gitbook-docs/rest_api.md

```diff
@@ -65,3 +65,9 @@ of the data under the `vessels` container in JSON format. Likewise, `GET
 /signalk/v1/api/vessels/urn:mrn:signalk:uuid:c0d79334-4e25-4245-8892-54e8ccc8021d` should return data for one specific
 vessel. In other words, the full Signal K data model SHOULD be traversable by any client making GET requests to an
 arbitrary depth.
+
+## History snapshot retrieval
```

Comment:
As a suggestion to debate, how about making the URL contain the timestamp, this way it would be a bit more RESTful:

```
GET /signalk/v1/snapshot/2018-08-24T15:19:09Z/vessels/self
```

Reply:
Did not think about this option. Using the same url as normally with an extra parameter makes client implementation maybe more straightforward and the query parameter can be thought of as a qualifier: 'i want resource X, with this twist'. A separate prefix may be more explicit and cleaner in that way.

Any other APIs we could draw experience from?

Reply:
This makes things more complicated on the client side. Much easier to add a parameter than having to much with the path.

Reply:
True, and existing server will probably just ignore the query parameter and return he current data, so it would look like they support it.

Reply:
Resolved in a5c4d504aa86b867be730977dae35a97dc2e59f7